File: src/net/liplum/components/SharingHealth.java
package net.liplum.components;

import net.liplum.blocks.StatedWall;
import mindustry.gen.Building;

public class SharingHealth extends ComponentBase<StatedWall.StatedWallBuild> {
    @Override
    public void onUpdate(StatedWall.StatedWallBuild b) {
        for (Building other : b.proximity) {
            if (other.block != b.block)
                continue;
            float thisH = b.health;
            float otherH = other.health;
            if (thisH > otherH && thisH > 1 && otherH < other.maxHealth) {
                b.health -= 1;
                other.health += 1;
            }
        }
    }
}

File: src/net/liplum/components/SharingHealth.kt
package net.liplum.components

import net.liplum.blocks.StatedWall.StatedWallBuild

object SharingHealth : ComponentBase<StatedWallBuild>() {
    override fun onUpdate(b: StatedWallBuild) {
        for (other in b.proximity) {
            if (other.block !== b.block) continue
            val thisH = b.health
            val otherH = other.health
            if (thisH > otherH && thisH > 1 && otherH < other.maxHealth) {
                b.health -= 1f
                other.health += 1f
            }
        }
    }
}